### PR TITLE
Enable loading stitched tests in partition mode

### PR DESCRIPTION
**Pull request by junjiemao1**
This series enables the execution of acrn unit tests atop a partition-mode-configured ACRN hypervisor. The test cases can be stitched together and selected by the command line passed via multiboot protocol. It is thus possible to control the execution and collect the results of multiple test cases by talking to the HV shell provided that VM creation/destruction commands are provided.

Summary of changes:

1. Fixes: segment selector reloading, .bss initialization and .comment/.note discarding
2. Tweaks: binary loading address adjustment
3. Enabling: Raw binary generation and test case stitching

Signed-off-by: Junjie Mao <junjie.mao@intel.com>

### CODE REVIEW

**Comment by junjiemao1** [comment]
@fyin1 Please help review this series. Thanks!

File: guest/x86/flat.lds [hunk 1]
```diff
@@ -1,8 +1,9 @@
 SECTIONS
 {
-    . = 4M + SIZEOF_HEADERS;
+    . = 4M;
```
Comment (fyin1-zz) [inline guest/x86/flat.lds]: Can we make the entry to 4M + SIZEOF_HEADERS + 0xC?

Comment (fyin1-zz) [reply guest/x86/flat.lds]: Other changes look good to me.

Comment (junjiemao1) [reply guest/x86/flat.lds]: SIZEOF_HEADERS may change from time to time (e.g. due to the number of sections in the linked executable). This makes the identification of the entry address impossible as we use an objdump'ed raw binary for partition mode.

How about we have two lds, one for single-test executable that should to be loaded as an ELF and the other for the stitched executable that have a fixed entry address and can be loaded as a raw binary?

Comment (fyin1-zz) [reply guest/x86/flat.lds]: Two lds is complicated. Can we have a header added at the beginning of each raw binary which include the binary entry in fixed offset?

Comment (junjiemao1) [reply guest/x86/flat.lds]: Sorry for the late response.

I have restored the original placement of sections. We can rely on the multiboot header which is always 12-byte and right ahead of the entry point.

Also the .comment and .note sections are discarded as it has been met that ignoring these sections in the linker script leads to a very small bss_start (mostly <1K), which leads the test program to crash when it tries to initialize its .bss section.

**Review by fyin1-zz — APPROVED** [review]

**Pull request merged**